webui-src/app/chat/chat.js

```diff
@@ -108,69 +183,184 @@ const ChatLobbyModel = {
+    },
+    view: () => m('.message', m('span.datetime', datetime), m('span.username', username), m('span.messagetext', text))
+  };
+}
+
+
 const Lobby = () => {
   let info = {};
+  let tagname = '';
+  let onclick = e => {};
+  let lobbytagname = '';
   return {
-    oninit: (v) => info = v.attrs.info,
-    view: (v) => m( '.lobby.' + (ChatRoomsModel.subscribed(v.attrs.info) ? 'subscribed':'public'), {
-      key: v.attrs.info.lobby_id.xstr64,
-      onclick: e => {
-        if (ChatRoomsModel.subscribed(v.attrs.info)) {
-          m.route.set('/chat/:lobby', {
-            lobby: v.attrs.info.lobby_id.xstr64
-          });
-        }
-      },
+    oninit: (v) => {
+        info = v.attrs.info;
+        tagname = v.attrs.tagname;
+        onclick = v.attrs.onclick || (e => {});
+        lobbytagname= v.attrs.lobbytagname || 'mainname';
+    },
+    view: v => {
+      return m(ChatLobbyModel.selected(info, '.selected-lobby', tagname), {
+      key: info.lobby_id.xstr64,
+      onclick: onclick,
     }, [
-      m('h5', info.lobby_name === '' ? '<unnamed>' : info.lobby_name),
-      m('p', info.lobby_topic),
-    ]),
+      m('h5', {class: lobbytagname},info.lobby_name === '' ? '<unnamed>' : info.lobby_name),
+      m('.topic', info.lobby_topic),
+    ]);
+    },
   };
 };
 
-const SubscribedLobbies = () => {
-  lobbies = [];
-  return {
-    oninit: (v) => ChatRoomsModel.loadSubscribedRooms(),
-    view: () => m('.widget', [
-      m('h3', 'Subscribed chat rooms'),
-      m('hr'),
-      Object.values(ChatRoomsModel.subscribedRooms).map(info => m(Lobby, {
-        info
-      })),
-    ]),
-  };
-};
+const LobbyList = () => {
+    return {
+      view: vnode => {
+        rooms = vnode.attrs.rooms;
+        tagname = vnode.attrs.tagname;
+        lobbytagname= vnode.attrs.lobbytagname;
+        onclick=vnode.attrs.onclick||(() => null);
+        return [
+          m('hr'),
+          rooms.map(info => m(Lobby, {
+            info,
+            tagname: tagname,
+            lobbytagname,
+            onclick: onclick(info),
+          }))
+        ];
+      },
+    };
+}
+
+const SubscribedLeftLobbies = () => [
+  m('h5.lefttitle', 'subscribed:'),
+  m(LobbyList, {
+    rooms: sortLobbies(Object.values(ChatRoomsModel.subscribedRooms)),
+    tagname: '.leftlobby.subscribed',
+    lobbytagname:'leftname',
+    onclick: ChatLobbyModel.switchToEvent,
+  })
+];
+
+const SubscribedLobbies = () => (
+  m('.widget', [
+    m('h3', 'Subscribed chat rooms'),
+    m(LobbyList, {
+      rooms: sortLobbies(Object.values(ChatRoomsModel.subscribedRooms)),
+      tagname:'.lobby.subscribed',
+      onclick: ChatLobbyModel.switchToEvent,
+    }),
+  ])
+);
+
+const PublicLeftLobbies = () => [
+  m('h5.lefttitle', 'public:'),
+  m(LobbyList, {
+    rooms: Object.values(ChatRoomsModel.allRooms).filter(info => !ChatRoomsModel.subscribed(info)),
+    tagname: '.leftlobby.public',
+    lobbytagname: 'leftname',
+    onclick: ChatLobbyModel.setupEvent,
+  }),
+];
```

Comment:
We should stick to using components whenever possible.

This is because mithril is clever and can optimize views for faster rendering.
But using functions will create new component on every render, making it less efficient.

See the anti patterns list: https://mithril.js.org/components.html#avoid-anti-patterns
Especially this section: https://mithril.js.org/components.html#define-components-statically,-call-them-dynamically

Reply:
it's a little bit tricky to find out, which parts can move to components

Reply:
You could've also simply converted them to closure components. I should have mentioned it lol.
But your changes are also fine, I don't see any problem with it.